src/sentinel.c

```diff
@@ -2175,7 +2175,11 @@ void rewriteConfigSentinelOption(struct rewriteConfigState *state) {
             line = sdscatprintf(sdsempty(),
                 "sentinel known-replica %s %s %d",
                 master->name, announceSentinelAddr(slave_addr), slave_addr->port);
-            rewriteConfigRewriteLine(state,"sentinel known-replica",line,1);
+            /* try to replace any known-slave option first if found */
+            if(rewriteConfigRewriteLine(state,"sentinel known-slave",line,0) ==0){
+                rewriteConfigRewriteLine(state, "sentinel known-replica", line, 1);
+            }
+            sdsfree(line);
```

Comment:
I think you are having double free of memory. Function `rewriteConfigRewriteLine()`
is the one to release `line`. I think you should do instead:
```
  if (rewriteConfigRewriteLine(state,"sentinel known-slave",sdsdup(line),0) == 0) {
    rewriteConfigRewriteLine(state, "sentinel known-replica", line, 1);
  } else {
    sdsfree(line);
 }
```
I didn't suggest to write automated test. The least we can do is to test it 
manually throughly that it is working as expected. Please verify that:
1. if having a configuration file with "known-slave" then the configuration file 
    after update will only include "known-replica".
3.  And if having a configuration with "known-replica" then configuration file
   after update is having expected data with only "known-replica".

Thank you

Reply:
Thank you @moticless :)